lib/oga/xml/parser.rll

```diff
@@ -245,7 +245,7 @@ string_body
   def reset
     @line = 1
 
-    @lexer.reset
+    @lexer.reset_native
   end
```

Comment:
@YorickPeterse Removing this causes 6 tests to fail. I just pushed [this change](https://github.com/YorickPeterse/oga/pull/154/commits/08df2d2340fc7bdbe36b14cbf52ae96bd7451863) so you can see the tests failures on [Travis CI](https://travis-ci.org/YorickPeterse/oga/builds/143612168). Here is a snippet of the output:

```
Failures:

  1) Oga::XML::PullParser tracking nodes resets the current node after parsing
     Failure/Error: nesting << @node.name
     
     NoMethodError:
       undefined method `<<' for nil:NilClass
     # ./lib/oga/xml/pull_parser.rb:142:in `on_element'
     # ./lib/oga/xml/parser.rb:491:in `_rule_27'
     # ./lib/oga/xml/parser.rb:232:in `block in each_token'
     # ./lib/oga/xml/lexer.rb:219:in `add_token'
     # ./lib/oga/xml/lexer.rb:421:in `on_element_end'
     # ./lib/oga/xml/lexer.rb:172:in `advance_native'
     # ./lib/oga/xml/lexer.rb:172:in `block in advance'
     # ./lib/oga/xml/lexer.rb:126:in `read_data'
     # ./lib/oga/xml/lexer.rb:171:in `advance'
     # ./lib/oga/xml/parser.rb:229:in `each_token'
     # ./lib/oga/xml/pull_parser.rb:72:in `parse'
     # ./lib/oga/xml/pull_parser.rb:72:in `parse'
     # ./spec/oga/xml/pull_parser/general_spec.rb:16:in `block (3 levels) in <top (required)>'

  2) Oga::XML::PullParser tracking nodes tracks the current node
     Failure/Error: raise LL::ParserError, message
     
     LL::ParserError:
       Unexpected end of input, expected element closing tag instead on line 1
     # ./lib/oga/xml/parser.rb:257:in `parser_error'
     # ./lib/oga/xml/parser.rb:235:in `each_token'
     # ./lib/oga/xml/pull_parser.rb:72:in `parse'
     # ./lib/oga/xml/pull_parser.rb:72:in `parse'
     # ./spec/oga/xml/pull_parser/general_spec.rb:10:in `block (3 levels) in <top (required)>'

  3) Oga::XML::PullParser elements parses the text of an element
     Failure/Error: nesting << @node.name
     
     NoMethodError:
       undefined method `<<' for nil:NilClass
     # ./lib/oga/xml/pull_parser.rb:142:in `on_element'
     # ./lib/oga/xml/parser.rb:491:in `_rule_27'
     # ./lib/oga/xml/parser.rb:232:in `block in each_token'
     # ./lib/oga/xml/lexer.rb:219:in `add_token'
     # ./lib/oga/xml/lexer.rb:431:in `on_text'
     # ./lib/oga/xml/lexer.rb:172:in `advance_native'
     # ./lib/oga/xml/lexer.rb:172:in `block in advance'
     # ./lib/oga/xml/lexer.rb:126:in `read_data'
     # ./lib/oga/xml/lexer.rb:171:in `advance'
     # ./lib/oga/xml/parser.rb:229:in `each_token'
     # ./lib/oga/xml/pull_parser.rb:72:in `parse'
     # ./lib/oga/xml/pull_parser.rb:72:in `parse'
     # ./spec/oga/xml/pull_parser/element_spec.rb:22:in `block (3 levels) in <top (required)>'

  4) Oga::XML::PullParser elements parses an element
     Failure/Error: raise LL::ParserError, message
     
     LL::ParserError:
       Unexpected end of input, expected element closing tag instead on line 1
     # ./lib/oga/xml/parser.rb:257:in `parser_error'
     # ./lib/oga/xml/parser.rb:235:in `each_token'
     # ./lib/oga/xml/pull_parser.rb:72:in `parse'
     # ./lib/oga/xml/pull_parser.rb:72:in `parse'
     # ./spec/oga/xml/pull_parser/element_spec.rb:12:in `block (3 levels) in <top (required)>'

  5) Oga::XML::PullParser tracking element nesting sets the nesting for the outer element
     Failure/Error: nesting << @node.name
     
     NoMethodError:
       undefined method `<<' for nil:NilClass
     # ./lib/oga/xml/pull_parser.rb:142:in `on_element'
     # ./lib/oga/xml/parser.rb:491:in `_rule_27'
     # ./lib/oga/xml/parser.rb:232:in `block in each_token'
     # ./lib/oga/xml/lexer.rb:219:in `add_token'
     # ./lib/oga/xml/lexer.rb:380:in `add_element'
     # ./lib/oga/xml/lexer.rb:351:in `on_element_name'
     # ./lib/oga/xml/lexer.rb:172:in `advance_native'
     # ./lib/oga/xml/lexer.rb:172:in `block in advance'
     # ./lib/oga/xml/lexer.rb:126:in `read_data'
     # ./lib/oga/xml/lexer.rb:171:in `advance'
     # ./lib/oga/xml/parser.rb:229:in `each_token'
     # ./lib/oga/xml/pull_parser.rb:72:in `parse'
     # ./lib/oga/xml/pull_parser.rb:72:in `parse'
     # ./spec/oga/xml/pull_parser/element_nesting_spec.rb:10:in `block (3 levels) in <top (required)>'

  6) Oga::XML::PullParser tracking element nesting pops element names after leaving an element
     Failure/Error: nesting << @node.name
     
     NoMethodError:
       undefined method `<<' for nil:NilClass
     # ./lib/oga/xml/pull_parser.rb:142:in `on_element'
     # ./lib/oga/xml/parser.rb:491:in `_rule_27'
     # ./lib/oga/xml/parser.rb:232:in `block in each_token'
     # ./lib/oga/xml/lexer.rb:219:in `add_token'
     # ./lib/oga/xml/lexer.rb:380:in `add_element'
     # ./lib/oga/xml/lexer.rb:351:in `on_element_name'
     # ./lib/oga/xml/lexer.rb:172:in `advance_native'
     # ./lib/oga/xml/lexer.rb:172:in `block in advance'
     # ./lib/oga/xml/lexer.rb:126:in `read_data'
     # ./lib/oga/xml/lexer.rb:171:in `advance'
     # ./lib/oga/xml/parser.rb:229:in `each_token'
     # ./lib/oga/xml/pull_parser.rb:72:in `parse'
     # ./lib/oga/xml/pull_parser.rb:72:in `parse'
     # ./spec/oga/xml/pull_parser/element_nesting_spec.rb:20:in `block (3 levels) in <top (required)>'

Finished in 0.75791 seconds (files took 0.53493 seconds to load)
1554 examples, 6 failures

Failed examples:

rspec ./spec/oga/xml/pull_parser/general_spec.rb:15 # Oga::XML::PullParser tracking nodes resets the current node after parsing
rspec ./spec/oga/xml/pull_parser/general_spec.rb:9 # Oga::XML::PullParser tracking nodes tracks the current node
rspec ./spec/oga/xml/pull_parser/element_spec.rb:19 # Oga::XML::PullParser elements parses the text of an element
rspec ./spec/oga/xml/pull_parser/element_spec.rb:9 # Oga::XML::PullParser elements parses an element
rspec ./spec/oga/xml/pull_parser/element_nesting_spec.rb:9 # Oga::XML::PullParser tracking element nesting sets the nesting for the outer element
rspec ./spec/oga/xml/pull_parser/element_nesting_spec.rb:17 # Oga::XML::PullParser tracking element nesting pops element names after leaving an element
```